cli/azd/pkg/input/console.go

```diff
@@ -63,14 +67,35 @@ func (c *AskerConsole) SetWriter(writer io.Writer) {
 
 // Prints out a message to the underlying console write
 func (c *AskerConsole) Message(ctx context.Context, message string) {
-	// Only write to the console during interactive & non-formatted responses.
-	if c.interactive && (c.formatter == nil || c.formatter.Kind() == output.NoneFormat) {
+	if c.formatter != nil && c.formatter.Kind() == output.JsonFormat {
```

Comment:
After talking a bit with @weikanglim I had tried a larger refactoring which tried to move all the calls to `.Message()` out from the Provider, in favor of emitting events which the `cmd` layer would interpret and format.  But I kept getting lost in the weeds doing that, so I ended up with this sort of hacky thing.

I do think that the way the system works today we get the correct /output/ written and so longer term we can try to refactor things such that the output is done at a much higher level, if we wanted.

This allows VS to show basically the same thing that VS Code would (since they just run us in the integrated terminal).  The only difference is this output is not colorized (which seems much more reasonable when trying to consume stuff programmatically).

Reply:
Behaviorally this is the right approach. I agree that the current way we're handling output is a mess, and we'll need refactoring steps to slowly untangle. If we get the right behavior, I'm less concerned about the internal structure for thje time being